patches/tModLoader/Terraria/ModLoader/ModTile.cs

```diff
@@ -415,6 +415,27 @@ public virtual void DrawEffects(int i, int j, SpriteBatch spriteBatch, ref TileD
 	{
 	}
 
+	/// <summary>
+	/// Used to spawn Dust or Gore particle effects.
+	/// Note that this is called even if the tile is invisible due to echo coating, so checking <paramref name="visible"/> should be used if dust should not be spawned unless the tile is visible. Tiles that still spawn particle effects while invisible can be useful to builders. Some tiles that spawn dust even when invisible include BubbleMachine, FogMachine, BrazierSuspended, Campfire, Chimney, SillyBalloonMachine, LeafBlock, and PoopBlock.
+	///
+	/// <para/> The <paramref name="tileFrameX"/> and <paramref name="tileFrameY"/> values differ from the Tile frame values in that they incorporate the changes from <see cref="SetDrawPositions"/> and likely should be used instead of <see cref="Tile.TileFrameX"/> and Y.
+	///
+	/// Returns true by default. Return false to stop the game from running code for <see cref="Main.tileShine"/> and shooting star dust while in near the shimmer.
+	/// </summary>
+	/// <param name="i"></param>
+	/// <param name="j"></param>
+	/// <param name="tileCache"></param>
+	/// <param name="tileFrameX"></param>
+	/// <param name="tileFrameY"></param>
+	/// <param name="tileLight"></param>
+	/// <param name="visible"></param>
+	/// <returns></returns>
+	public virtual bool DrawTilesEmitParticles(int i, int j, Tile tileCache, short tileFrameX, short tileFrameY, Color tileLight, bool visible)
```

Comment:
Add docs to `DrawEffects` to tell modders to use this method instead to emit particles (and why)

Reply:
@JavidPack reminder for this one